src/types/mod.rs

```diff
@@ -2670,3 +2670,116 @@ mod transaction_fee_distribution {
         }
     }
 }
+
+pub mod peers_info {
+    use super::*;
+    use crate::v2::Require;
+    use std::net::{IpAddr, SocketAddr};
+
+    // A vector of the peers that
+    // the node is connected to.
+    #[derive(Debug)]
+    pub struct PeersInfo {
+        pub peers: Vec<Peer>,
+    }
+
+    /// A peer that the node is connected to
+    #[derive(Debug)]
+    pub struct Peer {
+        // The id of the peer.
+        pub peer_id:        String,
+        // Type of the peer.
+        pub peer_type:      PeerType,
+        // Catchup status of the peer.
+        pub catchup_status: CatchupStatus,
```

Comment:
I think these two should be merged into an enum so that we don't have catchupstatus for bootstrappers.